src/utils.rs

```diff
@@ -20,10 +22,34 @@ pub fn time_as_num(time: SystemTime) -> f64 {
     (time.duration_since(UNIX_EPOCH).unwrap().as_millis() as f64) / 1000.0
 }
 
+/// Write a byte slice to disk.
+///
+/// Succeeds if the file already exists with identical contents.
+/// On the other hand, an AlreadyExists error is returned if the file exists with different contents.
+pub fn write_file_idempotent(path: &Path, contents: &[u8]) -> io::Result<()> {
+    match fs::OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(path)
+    {
+        Ok(mut f) => {
+            f.write_all(contents)?;
+        }
+        Err(err) if err.kind() == io::ErrorKind::AlreadyExists => {
+            if fs::read(path)? != contents {
+                return Err(err);
+            }
```

Comment:
Are we ok with end-of-line standardisartion between windows/everyone else here? I can't remember what we do in this case - perhaps we dodge it by spitting everything out on one line first?

Reply:
outpack_server doesn't ever do any special processing of newline or anything else. Metadata files are an immutable blob of bytes.